### PR TITLE
Feature/5743 authentication keystone

### PR DESCRIPTION
**Pull request by jesuspg**
@jframos @flopezag

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+/**
+ * Program configuration.
+ *
+ * <var>config</var> attributes will be updated with those read from <var>configFile</var>, if exists,
+ * and from command line arguments, in that order.
+ */
+var config = {
+    configFile: path.join(__dirname, '/../config', name + '.yml'),
+    listenPort: 3000,
```
Comment (flopezag) [inline lib/config.js]: It is not the app:port that you have in the aiakos.yml?

File: lib/routes/openstack.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+                callback();
+            }
+        });
+
+    });
+    req.on('error', function (e) {
+        if (e.code === 'ECONNRESET') {
+            console.log('Error in connection by TIMEOUT with keystone: ' + e);
+        } else {
+            console.log('Error in connection with keystone: ' + e);
```
Comment (flopezag) [inline lib/routes/openstack.js]: What do you do if the admin token expired or the admin token is wrong token?

File: test/unit/test_aiakos.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+#----------------------------------------------------------------------------
+app:
+  port: 3000		# App listen port
+
+#-----------------------------------------------------------------------------
+#  openstack admin (mandatory)
+#-----------------------------------------------------------------------------
+openstack:
+  username: ''		# admin openstack user
+  password: ''		# admin openstack password
```
Comment (flopezag) [inline test/unit/test_aiakos.yml]: You cannot use fake values in the username and password to check them in the unit tests?

File: lib/config.js (hunk 1 above)
Comment (jesuspg) [inline lib/config.js]: yes, but, this is a default value

Comment (flopezag) [inline lib/config.js]: NTC

File: test/unit/test_aiakos.yml (hunk 3 above)
Comment (jesuspg) [inline test/unit/test_aiakos.yml]: this file is only used by test_config.js, in order to test, how to config.js reads the file.

Comment (flopezag) [inline test/unit/test_aiakos.yml]: NTC

File: lib/routes/openstack.js [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+function getValidateToken(adminToken, token, callback) {
+    var headers = {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+        'X-Auth-Token': adminToken,
+        'X-Subject-Token': token
+    };
+
+    var options = {
+        host: 'cloud.lab.fiware.org',
```
Comment (jframos) [inline lib/routes/openstack.js]: These data should come from configuration file.

File: lib/routes/openstack.js [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+    var payloadString = JSON.stringify(payload);
+    var headers = {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+        'Content-Length': payloadString.length
+    };
+
+    var options = {
+        host: 'cloud.lab.fiware.org',
```
Comment (jframos) [inline lib/routes/openstack.js]: These data should come from configuration file.

File: lib/routes/v1.js [hunk 6]
```diff
@@ -102,6 +103,47 @@ function discoverKey(content) {
...
+                res.setHeader('Content-Length', content.length);
+                res.write(content, 'utf8');
+                res.end();
+
+            });
+        }
+
+
+
+
```
Comment (jframos) [inline lib/routes/v1.js]: Is the code passing all _jshint_ checks?

File: lib/routes/openstack.js [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+ */
+'use strict';
+
+var http = require('http'),
+    config = require('../config').data;
+
+
+/**
+ * @function parseRegionNameFromUserName
+ * Get region name from username. username MUST be admin-regionName
```
Comment (jframos) [inline lib/routes/openstack.js]: You have to consider that some of the admin names do not have the "real" name of the region. Do you remember the problem we had with Dashboard authorization?

File: lib/routes/v1.js (hunk 6 above)
Comment (jesuspg) [inline lib/routes/v1.js]: yes, you could see here: https://travis-ci.org/telefonicaid/fiware-aiakos/builds/97618645#L402

File: lib/routes/openstack.js (hunk 7 above)
Comment (jesuspg) [inline lib/routes/openstack.js]: please, review the last commit

File: lib/routes/openstack.js (hunk 2 above)
Comment (flopezag) [inline lib/routes/openstack.js]: NTC

File: test/unit/test_openstack.js [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+            //then
+            assert(true);
+        }
+
+
+    });
+
+    test('should_parse_region_name_and_match_with_configuration', function() {
+        //given
+        config.regions.names = [{'admin-regionName': 'region1'}];
```
Comment (flopezag) [inline test/unit/test_openstack.js]: add a specific case 'admin-cityname-partner': 'anyregionname' ...

File: lib/routes/v1.js (hunk 6 above)
Comment (flopezag) [inline lib/routes/v1.js]: NTC

**Pull request merged**